src/confidentialDataRule.cpp

```diff
@@ -0,0 +1,51 @@
+#include "confientialDataRule.h"
+#include "filesHandler.h"
+
+bool checkIfMaliciousIp(const std::string& dstIp)
+{
+	if (isStrExistsInFile(maliciousIpsFile, dstIp) != "0")
+		return true;
+	return false;
+}
+
+bool checkIfDataConfidential(const std::string& data)
+{
+	std::vector <std::string> tokens;
+
+	int countTotalWords = 0;
+	int countBadWords = 0;
+	std::stringstream check(data);
+
+	std::string intermediate;
+
+
+	while (getline(check, intermediate, ' '))
+	{
+		tokens.push_back(intermediate);
+	}
+	int num = 0;
+
+	for (int i = 0; i < tokens.size(); i++)

```

Comment:
about which shift to you mean?
I call to a function with each word because I need to check if the word exists in a file and if it does then to extract the value.

Reply:
>>

Reply:
this is not a shift, 
strToInt is a stream and if you want to add to the stream a value then you use >>
this is how streams handle in cpp